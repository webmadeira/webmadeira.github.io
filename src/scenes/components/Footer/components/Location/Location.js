import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { H3, Body2 } from '../../../../../components/typography'

const LocationContainer = styled.div`
  text-align: left;
  margin-right: 8px;
`

const LocationName = styled.a`
  color: ${props => props.theme.color('neutral', 'bright')}
  margin-top: 14px;
`

const LocationDetails = styled.div`
  margin-top: 15px;
`

const Location = ({ location }) => (
  <LocationContainer>
    <H3>LOCATION</H3>
    <LocationDetails>
      <LocationName href={location.link} target="_black" rel="noopener">
        <Body2>{location.name}</Body2>
      </LocationName>
      <Body2>{location.address}</Body2>
      <Body2>{location.postCode}</Body2>
    </LocationDetails>
  </LocationContainer>
)

Location.propTypes = {
  location: PropTypes.shape({
    name: PropTypes.string,
    address: PropTypes.string,
    postCode: PropTypes.string,
  }).isRequired,
}

export default Location
