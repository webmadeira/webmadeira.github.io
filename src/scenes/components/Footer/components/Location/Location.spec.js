import React from 'react'
import { shallow } from 'enzyme'

import Location from './Location'

describe('Location component', () => {
  it('must render properly', () => {
    const location = {
      name: 'test name',
      address: 'test address',
      postCode: 'test post code',
    }

    const component = shallow(<Location location={location} />)

    expect(component).toMatchSnapshot()
  })
})
