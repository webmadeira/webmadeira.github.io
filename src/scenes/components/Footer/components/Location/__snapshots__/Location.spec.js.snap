// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Location
    location={
      Object {
        "address": "test address",
        "name": "test name",
        "postCode": "test post code",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.h3>
          LOCATION
        </styled.h3>,
        <styled.div>
          <styled.a
            href={undefined}
            rel="noopener"
            target="_black"
          >
            <styled.p>
              test name
            </styled.p>
          </styled.a>
          <styled.p>
            test address
          </styled.p>
          <styled.p>
            test post code
          </styled.p>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "LOCATION",
        },
        "ref": null,
        "rendered": "LOCATION",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.a
              href={undefined}
              rel="noopener"
              target="_black"
            >
              <styled.p>
                test name
              </styled.p>
            </styled.a>,
            <styled.p>
              test address
            </styled.p>,
            <styled.p>
              test post code
            </styled.p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.p>
                test name
              </styled.p>,
              "href": undefined,
              "rel": "noopener",
              "target": "_black",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test name",
              },
              "ref": null,
              "rendered": "test name",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "test address",
            },
            "ref": null,
            "rendered": "test address",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "test post code",
            },
            "ref": null,
            "rendered": "test post code",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.h3>
            LOCATION
          </styled.h3>,
          <styled.div>
            <styled.a
              href={undefined}
              rel="noopener"
              target="_black"
            >
              <styled.p>
                test name
              </styled.p>
            </styled.a>
            <styled.p>
              test address
            </styled.p>
            <styled.p>
              test post code
            </styled.p>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "LOCATION",
          },
          "ref": null,
          "rendered": "LOCATION",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.a
                href={undefined}
                rel="noopener"
                target="_black"
              >
                <styled.p>
                  test name
                </styled.p>
              </styled.a>,
              <styled.p>
                test address
              </styled.p>,
              <styled.p>
                test post code
              </styled.p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <styled.p>
                  test name
                </styled.p>,
                "href": undefined,
                "rel": "noopener",
                "target": "_black",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "test name",
                },
                "ref": null,
                "rendered": "test name",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test address",
              },
              "ref": null,
              "rendered": "test address",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "test post code",
              },
              "ref": null,
              "rendered": "test post code",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
